sailor/sap_iot/write.py

```diff
@@ -86,4 +128,9 @@ def upload_indicator_data(dataset: TimeseriesDataset):
 
     for (group_id, template_id), group_indicators in query_groups.items():
         selected_indicator_set = ac_indicators.IndicatorSet(group_indicators)
+
+        if force_update == '':
+            uploaded_indicators = list(selected_indicator_set.as_df()['name'])
```

Comment:
don't transform to a dataframe first here.
something like 
`uploaded_indicators = [indicator.id for indicator in selected_indicator_set]`
seems more appropriate